### ISSUE TITLE
Transfer encoding in HTTP headers

### ISSUE THREAD

**Issue by danielstocks** [bitinn/node-fetch#47]
Hello,

It seems that window.fetch in the browser will treat `Content-Length` and `Transfer-Encoding` differently compared to node-fetch. This causes some headache as a receiving server will have to treat these results differently (stream vs no stream).

This is how I'm calling fetch (both on the client and the server)
```js
fetch("http://localhost:5000/users", { method: "POST" });
```

Using node-fetch in Node 4.1 yields the following headers on the receiving server:
```
Transfer-Encoding: chunked
Content-Length:
```

Where as using `window.fetch` in Chrome 45.0.2454.85 produces the following:
```
Content-Length: 0
```

I'm not sure about this, but my guess is that  `Transfer-Encoding: chunked`should only be applied in the cases where you're providing a stream as body. As in this instance: https://github.com/bitinn/node-fetch/blob/master/test/test.js#L436

Otherwise, it should set the `content-length` header. I.e something like this:
```js
options.headers['Content-Length'] = Buffer.byteLength(body);
```

I may have misunderstood the spec, so let me know if this fits within the scope of this project or not. If so, I'm happy to submit a PR.

Regards,
Daniel

**Comment by bitinn** [comment]
I want to make sure I understand your problem:

- Are you saying node-fetch send out request with `Transfer-Encoding: chunked` by default? Since this is likely a default set by `http.request` (node.js http module), I wonder their reason behind it.

- Is this a problem for your project? Do you see anything in the Fetch Spec regarding this?

- Can you reproduce it on service like http://httpbin.org/headers, which should show you the request headers?

**Comment by danielstocks** [comment]
> Are you saying node-fetch send out request with Transfer-Encoding: chunked by default? Since this is likely a default set by http.request (node.js http module), I wonder their reason behind it.

Yes. It's a default set by the node `http.request` library. If you do set a `content-length` header however, the node `http.request` will **automatically remove** the 'Transfer-Encoding: chunked' header, so it's designed to send either of those. 

I don't know the reasoning behind it, but I know that many server protocols like WSGI (for running python apps) don't support `Transfer-Encoding: chunked` (https://github.com/mitsuhiko/flask/issues/367).

It makes sense that if you're not sending a stream, then you should be setting the `content-length' header, which node-fetch currently does not. It'll explicitly tell a server that it's not dealing with a stream. To be consistent with the current implementation of window.fetch in chrome, this would be the right thing do (although I don't know if they've implemented fetch correctly but I'd assume so)

Like I said. I can submit PR for it you agree on the above.

**Comment by bitinn** [comment]
Fixed in v1.3.3
